### PR TITLE
Corrections to character limit on ntnrc_client.dm

### PR DESCRIPTION
**Pull request by twilightwanderer**
## About The Pull Request

Corrects the chat restriction on using ASII characters only.

изображение

## Why It's Good For The Game

In the original code, the chat client uses only ASII standard characters, which is very limited in its capabilities. For example, does not allow you to use specialized characters, which would have taken the atmosphere of the old messengers, as well as regional characters. The lack of regional characters complicates the game for non-English-speaking servers. For example, the Russian-speaking player community Space Station 13 The Fluffy Frontier uses the original Skyrat build. And the players of this community almost do not use the chat client to communicate due to the fact that they can not use Cyrillic characters. 

I tried to solve this problem by rewriting some of the text handling functions, namely:
- /proc/reject_bad_message(text) - сhecks messages for emptiness;
- /proc/reject_bad_username(text, max_length = 32) - Checks the entered name for emptiness and length. (32 characters seems optimal to me for a nickname. But if not, it's easy to change.);
- /proc/reject_bad_chatname(text, max_length = 12) - Checks the entered name of the chat for emptiness and the number of characters (I chose 12 characters because a large number expands the panel with chats to the size in which you can not see the main message box). 

## Changelog
:cl:
add: Adding the ability to use a large number of characters in the chat client.
expansion: Added text module
code: In the chat client code, the standard text checking procedures have been replaced by modular procedures.
/:cl:

### CODE REVIEW

File: code/modules/modular_computers/file_system/programs/ntnrc_client.dm [hunk 1]
```diff
@@ -267,3 +267,58 @@
...
+			switch(text2ascii(char))
+				if(0 to 31)
+					return
+				if(32)
+					continue
+				else
+					non_whitespace = TRUE
+		if (non_whitespace)
+			return text
+//SKYRAT EDIT ADDITION END
```
Comment (GoldenAlpharex) [inline code/modules/modular_computers/file_system/programs/ntnrc_client.dm]: This should probably be put in another file in our `modular_skyrat` folder, they don't absolutely need to be there.

Comment (twilightwanderer) [reply code/modules/modular_computers/file_system/programs/ntnrc_client.dm]: Okay, I'll redo it soon.

Comment (twilightwanderer) [reply code/modules/modular_computers/file_system/programs/ntnrc_client.dm]: I put the procedures in a separate module on your advice. Should I put the [MODULAR] tag in the name of the PR?

**Comment by twilightwanderer** [comment]
The code has been updated. The created text checking procedures have been moved from ntnrc_client.dm to a separate module code\__HELPERS\~skyrat_helpers\text.dm

**Comment by Gandalf2k15** [comment]
Do this upstream.

**Comment by twilightwanderer** [comment]
> Do this upstream.

Okey. I'll do it soon.

**Comment by unit0016** [comment]
While you're here, @twilightwanderer - I've been meaning to get in contact with some of the fluffy frontier players to ask how radio frequencies we introduced (FACTION, Cybersun, Interdyne, The Enclave Pirates) are handled by non english keyboards, and if we'd need to do any work to make them work?
Thanks!

**Comment by twilightwanderer** [comment]
> While you're here, @twilightwanderer - I've been meaning to get in contact with some of the fluffy frontier players to ask how radio frequencies we introduced (FACTION, Cybersun, Interdyne, The Enclave Pirates) are handled by non english keyboards, and if we'd need to do any work to make them work?
> Thanks!

Thank you for asking. Everything seems to be working fine. I have tested and other players have not complained. If there are any problems with these frequencies, I will get back to you.

**Comment by GoldenAlpharex** [comment]
Since it's being done upstream and likely will get merged, I'll go ahead and close it for now.

**Pull request closed without merging**